$NetBSD$

- use @PREFIX@ in directory locations
- don't overwrite ${CFLAGS}
- use ${LDFLAGS} from pkgsrc
- install to ${DESTDIR}

--- Makefile.orig	2026-10-07 01:06:44.000000000 +0000
+++ Makefile
@@ -1,11 +1,11 @@
 #install paths
 #for linux, MacOS X, Cygwin:
-BIN=/usr/local/bin
-LIB=/usr/local/lib
-MAN=/usr/local/share/man/man1
+BIN=@PREFIX@/bin
+LIB=@PREFIX@/lib
+MAN=@PREFIX@/@PKGMANDIR@/man1
 #for Solaris:
 #MAN=/usr/local/man/man1
-CFLAGS = #-O #-DCYGWIN #-DUWIN #-DIBMRISC #-Dsparc7 #-Dsparc8
+CFLAGS += #-O #-DCYGWIN #-DUWIN #-DIBMRISC #-Dsparc7 #-Dsparc8
 #be wary of using anything higher than -O as the garbage collector may fall over
 #if using gcc rather than clang try without -O first
 EX = #.exe        #needed for CYGWIN, UWIN
@@ -23,7 +23,7 @@ all: FORCE
 # -Dsparc8 needed for Solaris 2.8 or later
 mira$(EX): big.o cmbnms.o data.o lex.o reduce.o signals.o steer.o trans.o types.o \
       utf8.o y.tab.o version.o $(LINENOISEOBJ) Makefile
-	$(CC) $(CFLAGS) -o mira$(EX) version.o cmbnms.o y.tab.o data.o lex.o \
+	$(CC) $(CFLAGS) $(LDFLAGS) -o mira$(EX) version.o cmbnms.o y.tab.o data.o lex.o \
 	    big.o reduce.o signals.o steer.o trans.o types.o utf8.o \
             $(LINENOISE) -lm
 
@@ -80,11 +80,11 @@ cleanup:
 install:
 	make -s all
 	strip mira$(EX)
-	cp mira$(EX) $(BIN)/
-	cp mira.1 $(MAN)/
-	rm -rf $(LIB)/miralib
+	cp mira$(EX) ${DESTDIR}/$(BIN)/
+	cp mira.1 ${DESTDIR}/$(MAN)/
+	rm -rf ${DESTDIR}/$(LIB)/miralib
 	strip miralib/menudriver$(EX)
-	cp -rp miralib $(LIB)/
+	cp -rp miralib ${DESTDIR}/$(LIB)/
 SOURCES = .xversion big.c big.h gencdecs data.h data.c lex.h lex.c reduce.c rules.y \
           signals.c signals.h  steer.c trans.c types.c utf8.h utf8.c version.c fdate.c
 sources: $(SOURCES); @echo $(SOURCES)
